asyncapi-cli/src/main/java/io/ballerina/asyncapi/cmd/AsyncApiCmd.java

```diff
@@ -124,6 +220,146 @@ public void execute() {
         }
     }
 
+    private void giveWarningsForInvalidSpecGenOptions() {
+        if (licenseFilePath != null) {
+            outStream.println(String.format(INVALID_OPTION_WARNING, LICENSE_FLAG, SPEC));
+        }
+        if (includeTestFiles) {
+            outStream.println(String.format(INVALID_OPTION_WARNING, TEST_FLAG, SPEC));
+        }
+    }
+
+    private void giveWarningsForInvalidClientGenOptions() {
+        if (generatedFileType) {
+            outStream.println(String.format(INVALID_OPTION_WARNING, JSON_FLAG, CLIENT));
+        }
+        if (service != null) {
+            outStream.println(String.format(INVALID_OPTION_WARNING, SERVICE_FLAG, CLIENT));
+        }
+    }
+
+    private void verifyValidInputsForHttp() {
+        if (licenseFilePath != null) {
+            outStream.println(String.format(INVALID_OPTION_ERROR_HTTP, LICENSE_FLAG));
+            exitError(this.exitWhenFinish);
+        }
+        if (service != null) {
+            outStream.println(String.format(INVALID_OPTION_ERROR_HTTP, SERVICE_FLAG));
+            exitError(this.exitWhenFinish);
+        }
+        if (includeTestFiles) {
+            outStream.println(String.format(INVALID_OPTION_ERROR_HTTP, TEST_FLAG));
+            exitError(this.exitWhenFinish);
+        }
+        if (generatedFileType) {
+            outStream.println(String.format(INVALID_OPTION_ERROR_HTTP, JSON_FLAG));
+            exitError(this.exitWhenFinish);
+        }
+    }
+
+    private void ballerinaToAsyncApiWs(String fileName) {
+        List<AsyncApiConverterDiagnostic> errors = new ArrayList<>();
+        final File balFile = new File(fileName);
+        Path balFilePath = null;
+        try {
+            balFilePath = Paths.get(balFile.getCanonicalPath());
+        } catch (IOException e) {
+            DiagnosticMessages message = DiagnosticMessages.AAS_CONVERTOR_102;
+            ExceptionDiagnostic error = new ExceptionDiagnostic(message.getCode(),
+                    message.getDescription(), null, e.getLocalizedMessage());
+            errors.add(error);
+        }
+        getTargetOutputPathWs();
+        // Check service name it is mandatory
+        BallerinaToAsyncApiGenerator asyncApiConverter = new BallerinaToAsyncApiGenerator();
```

Comment:
Do we need to initialize the async-api generator like this ? I assume the generator does not carry a state, in that case we can have the generator defined at the class level.
```java
private final BallerinaToAsyncApiGenerator asyncApiConverter = new BallerinaToAsyncApiGenerator();
```